apps/keyboard/js/keyboard/user_press_manager.js

```diff
@@ -234,11 +256,73 @@ UserPressManager.prototype._handleFinishPress = function(el, coords, id) {
   press.updateCoords(coords,
     press.moved || this._distanceReachesLimit(id, coords));
 
-  if (typeof this.onpressend === 'function') {
-    this.onpressend(press, id);
+  if(this._lowEndDevice && this._exceedSpeedLimit(press) &&
```

Comment:
Let's do

```javascript
if (this._isLowEndDevice) {
  this._handleLowEndDeviceFinishPress(...);
  return;
}
```

and move line 259 to line 292 into that function so people would know they are specifics to low end devices.